### PR TITLE
Content carousel support

### PR DESCRIPTION
**Pull request by izzyblues**
This PR includes: 

- Monitoring exclusion for publishes generated by Content Carousel
- Introduction of vendoring with govendor
- Added missing LICENSE file 👨‍⚖️

### CODE REVIEW

File: app_test.go [hunk 1]
```diff
@@ -34,10 +34,16 @@ func TestIsIgnorableMessage_naturalMessage(t *testing.T) {
 	}
 }
 
-func TestIsIgnorableMessage_syntheticMessage(t *testing.T) {
+func TestIsIgnorableMessageForMessagesToIgnore(t *testing.T) {
 	if !isIgnorableMessage(syntheticTID) {
 		t.Error("Synthetic message marked as normal")
 	}
+	if !isIgnorableMessage(carouselRepublishTID) {
+		t.Error("Courousel republish message marked as normal")
```
Comment (peteclark-ft) [inline app_test.go]: `Carousel` :)

File: app_test.go [hunk 2]
```diff
@@ -34,10 +34,16 @@ func TestIsIgnorableMessage_naturalMessage(t *testing.T) {
...
+	}
+	if !isIgnorableMessage(carouselGeneratedTID) {
+		t.Error("Courousel generated message marked as normal")
```
Comment (peteclark-ft) [inline app_test.go]: And again `Carousel`

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2016 Financial Times
```
Comment (khatton-ft) [inline LICENSE]: 2017

File: app.go [hunk 4]
```diff
@@ -305,9 +306,19 @@ func isMessagePastPublishSLA(date time.Time, threshold int) bool {
...
 
 func isIgnorableMessage(tid string) bool {
+	return isSyntheticTransactionID(tid) || isContentCarouselTransactionID(tid)
+}
+
+func isSyntheticTransactionID(tid string) bool {
 	return strings.HasPrefix(tid, "SYNTHETIC")
 }
 
+var carouselTransactionIDRegExp = regexp.MustCompile(`^(tid_[\S]+)_carousel_[\d]{10}.*$`)
```
Comment (khatton-ft) [inline app.go]: I'd prefer all the declarations at the top of the file. Also I wonder whether this should be `^(tid_[^_]+)_...`

**Pull request merged**